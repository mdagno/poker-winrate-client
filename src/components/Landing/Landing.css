.landing {
  margin-top: 5%;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  color: white;
  text-align: center;
}

.landingTitle {
  margin-top: 50px;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
  color: white;
  font-size: 2em;
}

.landing p {
  text-align: left;
  line-height: 2em;
  margin-bottom: 1.2em;
  margin-top: 1.2em;
  font-size: 1.08em;
}

.square-image {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
}

.rec-image {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
}

@media only screen and (min-device-width : 768px){
  .landing {
    width: 50%;
    margin-top: 15%;
  }
}

@media only screen and (min-device-width : 1024px){
  .landing {
    margin-top: 7%;
  }
}

