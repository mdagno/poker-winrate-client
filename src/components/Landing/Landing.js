import React from 'react';
import './Landing.css';
import UserContext from '../../ApiContext';
import imageOne from '../../app-screenshots/poker-winrate-one.PNG';
import imageTwo from '../../app-screenshots/poker-winrate-two.PNG';
import imageThree from '../../app-screenshots/poker-winrate-three.PNG';
import imageFour from '../../app-screenshots/poker-winrate-four.PNG';

export default class Landing extends React.Component {
  static contextType = UserContext; 
  
  render(){
    return(
      <div className='landing'>
      <h2 className='landingTitle'>Welcome to Poker WinRate!</h2>
      <p>Poker WinRate is a useful tool to determine your long term win rate in your poker games.  To improve your win rate overtime, it is critical to track your sessions over time to analyze your pattern of play over a long period of time to determine if your edge against other players is improving.</p>
      <p>
        1. Begin by creating an account and logging in.
      </p>
      <p>2. Add a completed sessions by clicking the "Add a Session" tab to access a for and fill in the required information.</p>
      <img src={imageOne} alt="Form to input session details" className="square-image"/>
      <p>3. Once added, past sessions can be accessed by clicking the "Your Sessions" tab.  A filter is available to sort though sessions by different game types.</p>
      <img src={imageThree} alt="All sessions rendered as indivual session cards" className="rec-image"/>
      <p>4. A detailed view of past sessions is available by clicking on one of the session cards.  In this view, a past session can be updated by clicking Edit.</p>
      <img src={imageFour} alt="Detailed view of a past session" className="rec-image"/>
      <p>5. A data visualization of all past sessions can be viewed by clicking the "Analytics" tab.</p>
      <img src={imageTwo} alt="Example graph of compiled session data" className="rec-image"/>
      </div>
    )
  }
}