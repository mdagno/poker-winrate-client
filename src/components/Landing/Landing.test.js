import React from 'react';
import Landing from './Landing';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

describe('Landing component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Landing />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Landing renders correctly', () => {
    const tree = renderer
      .create(<Landing />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
