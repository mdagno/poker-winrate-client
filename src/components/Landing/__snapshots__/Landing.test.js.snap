// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component Landing renders correctly 1`] = `
<div
  className="landing"
>
  <h2
    className="landingTitle"
  >
    Welcome to Poker WinRate!
  </h2>
  <p>
    Poker WinRate is a useful tool to determine your long term win rate in your poker games.  To improve your win rate overtime, it is critical to track your sessions over time to analyze your pattern of play over a long period of time to determine if your edge against other players is improving.
  </p>
  <p>
    1. Begin by creating an account and logging in.
  </p>
  <p>
    2. Add a completed sessions by clicking the "Add a Session" tab to access a for and fill in the required information.
  </p>
  <img
    alt="Form to input session details"
    className="square-image"
    src="poker-winrate-one.PNG"
  />
  <p>
    3. Once added, past sessions can be accessed by clicking the "Your Sessions" tab.  A filter is available to sort though sessions by different game types.
  </p>
  <img
    alt="All sessions rendered as indivual session cards"
    className="rec-image"
    src="poker-winrate-three.PNG"
  />
  <p>
    4. A detailed view of past sessions is available by clicking on one of the session cards.  In this view, a past session can be updated by clicking Edit.
  </p>
  <img
    alt="Detailed view of a past session"
    className="rec-image"
    src="poker-winrate-four.PNG"
  />
  <p>
    5. A data visualization of all past sessions can be viewed by clicking the "Analytics" tab.
  </p>
  <img
    alt="Example graph of compiled session data"
    className="rec-image"
    src="poker-winrate-two.PNG"
  />
</div>
`;
